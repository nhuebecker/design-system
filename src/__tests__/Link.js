import React from 'react'
import renderer from 'react-test-renderer'
import { Link } from '..'

describe('Link', () => {
  test('renders', () => {
    const json = renderer.create(<Link>Hello</Link>).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders underline', () => {
    const json = renderer.create(<Link underline>Underline</Link>).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('text-decoration', 'underline')
  })

  test('renders hoverline', () => {
    const json = renderer.create(<Link hoverline>Hoverline</Link>).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('text-decoration', 'none')
    expect(json).toHaveStyleRule('text-decoration', 'underline', {
      modifier: ':hover'
    })
  })
})
