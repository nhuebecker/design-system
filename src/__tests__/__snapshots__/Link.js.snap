// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
.c0 {
  color: #2d9ee4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:before,
.c0:after {
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  display: inline-block;
}

<a
  className="c0"
  color="info"
>
  Hello
</a>
`;

exports[`Link renders hoverline 1`] = `
.c0 {
  color: #2d9ee4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus,
.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:before,
.c0:after {
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  display: inline-block;
}

<a
  className="c0"
  color="info"
>
  Hoverline
</a>
`;

exports[`Link renders underline 1`] = `
.c0 {
  color: #2d9ee4;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:before,
.c0:after {
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  display: inline-block;
}

<a
  className="c0"
  color="info"
>
  Underline
</a>
`;
